cf-serverd/server_common.c

```diff
@@ -1736,37 +1736,42 @@ void SetConnIdentity(ServerConnectionState *conn, const char *username)
         memcpy(conn->username, username, username_len + 1);
     }
 
-#ifdef __MINGW32__            /* NT uses security identifier instead of uid */
-
     bool is_root = strcmp(conn->username, "root") == 0;
-
-    if (!NovaWin_UserNameToSid(conn->username, (SID *) conn->sid,
-                               CF_MAXSIDSIZE, !is_root))
-    {
-        memset(conn->sid, 0, CF_MAXSIDSIZE);  /* is invalid sid - discarded */
-    }
-
     if (is_root)
     {
-        /* It the remote user identifies himself as root, even on Windows
+        /* If the remote user identifies himself as root, even on Windows
          * cf-serverd must grant access to all files. uid==0 is checked later
          * in TranferRights() for that. */
         conn->uid = 0;
     }
 
-#else                                                 /* UNIX - common path */
+#ifdef __MINGW32__            /* NT uses security identifier instead of uid */
 
-    static pthread_mutex_t pwnam_mtx = PTHREAD_ERRORCHECK_MUTEX_INITIALIZER_NP;
-    struct passwd *pw = NULL;
+    if (!NovaWin_UserNameToSid(conn->username, (SID *) conn->sid,
+                               CF_MAXSIDSIZE, !is_root))
+    {
+        memset(conn->sid, 0, CF_MAXSIDSIZE);  /* is invalid sid - discarded */
+    }
 
-    if (ThreadLock(&pwnam_mtx))
+#else                                                 /* UNIX - common path */
+
+    if (conn->uid == CF_UNKNOWN_OWNER)      /* skip looking up UID for root */
```

Comment:
This is a performance optimisation for hubs, maybe we should leave it out of 3.6.x